### PR TITLE
Split files and big types

### PR DESCRIPTION
**Pull request by alexdupre**
This is a merge of my two other PRs plus an improved check on scripted tests.

### CODE REVIEW

**Review by blast-hardcheese — APPROVED** [review]
Tested on a real codebase, excellent!

File: docs/docs/client.html [hunk 1]
```diff
@@ -64,10 +64,12 @@
   def genView(value: Boolean): CalibanSettings                  // Provide a case class and helper method to select all fields on an object (default: false)
   def scalarMapping(mapping: (String,String)*): CalibanSettings // A mapping from GraphQL scalar types to JVM types, as unknown scalar types are represented as String by default.
   def imports(values: String*): CalibanSettings                 // A list of imports to be added to the top of a generated client
+  def splitFiles(value: Boolean): CalibanSettings               // Split single client object into multiple files (default: false)
```
Comment (ghostdogpr) [inline docs/docs/client.html:67]: Can you delete these changes?

**Review by ghostdogpr — COMMENTED** [review]
Looks great!

File: docs/docs/client.html (hunk 1 above)
Comment (alexdupre) [reply docs/docs/client.html]: Done

**Review by ghostdogpr — APPROVED** [review]
Great, thanks. I will cancel the doc generation step when I merge this so that docs aren't updated until the next release.

**Pull request merged**